### PR TITLE
Parameterize program and kernel info tests.

### PR DESCRIPTION
**Pull request by aarongreig**

### CODE REVIEW

**Review by veselypeta — APPROVED** [review]
LGTM!

File: test/conformance/program/urProgramGetInfo.cpp [hunk 1]
```diff
@@ -1,16 +1,30 @@
 // Copyright (C) 2023 Intel Corporation
 // SPDX-License-Identifier: MIT
 
+#include <map>
+
```
Comment (omarahmed1111) [inline test/conformance/program/urProgramGetInfo.cpp:5]: Do we need this include? (I just don't notice a usage for it in the file)

**Review by omarahmed1111 — APPROVED** [review]
LGTM, just a nit comment

**Review by kbenzie — APPROVED** [review]
LGTM

**Pull request merged**